lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -395,10 +394,11 @@ namespace utf8
             cch -= 4;
         }
 LSlowPath:
+        // WARNING: Assume cch correct, suppress end-of-buffer checking
+        // we might have failed to measure cch, +1 for odd length. sizeof(char16) == 2
+        LPCUTF8 end = ptr + cch + (cch % 2 ? 2 : 1);
         while (cch-- > 0)
         {
-            LPCUTF8 end = ptr + cch + 1; // WARNING: Assume cch correct, suppress end-of-buffer checking
```

Comment:
> I wonder if both before and after change are incorrect. 

I was looking into this with a random string generator. Looks like this algorithm is broken right from the beginning. Improving the test case and impl.